elfloader-tool/include/arch-arm/64/mode/structures.h

```diff
@@ -21,9 +21,9 @@
 #define PMD_BITS                9
 #define PMD_SIZE_BITS           (PMD_BITS + PMDE_SIZE_BITS)
 
-#define GET_PGD_INDEX(x)        (((x) >> (ARM_2MB_BLOCK_BITS + PMD_BITS + PUD_BITS)) & MASK(PGD_BITS))
-#define GET_PUD_INDEX(x)        (((x) >> (ARM_2MB_BLOCK_BITS + PMD_BITS)) & MASK(PUD_BITS))
-#define GET_PMD_INDEX(x)        (((x) >> (ARM_2MB_BLOCK_BITS)) & MASK(PMD_BITS))
+#define GET_PGD_INDEX(x)        (((word_t)(x) >> (ARM_2MB_BLOCK_BITS + PMD_BITS + PUD_BITS)) & MASK(PGD_BITS))
```

Comment:
In the commit comment it says that this works for pointer now also, but pointer must be cast to `uintptr_t` first to make then an integer. Maybe add this here also then?

Reply:
I believe the reason behind the comment is that in `tools/seL4/elfloader-tool/include/types.h` "word_t is practically an alias for size_t/uintptr_t on the platforms we support so far". Do you mean to just change `word_t` to `uintptr_t` here?

Reply:
I think strict C rules say, that pointers must be cast to `uintptr_t` to make them an integer, from there you can cast the integer to any other integer type.

Reply:
That is reasonable, I did find this excerpt in the C99 standard though, and it's the only part I could find on `uintptr_t`:

> The following type designates an unsigned integer type with the property that any valid
pointer to void can be converted to this type, then converted back to pointer to void,
and the result will compare equal to the original pointer: uintptr_t

So the strictest form would be a cast to `void *` first before `uintptr_t`. However, I think all the reasonable implementations have sensible ways for any pointer casted to `uintptr_t` to behave such that we can skip the `void *` cast.